mapper/orm/src/main/java/org/hibernate/search/mapper/orm/massindexing/MassIndexer.java

```diff
@@ -225,4 +225,16 @@ public interface MassIndexer {
 	@Incubating
 	MassIndexer failureFloodingThreshold(long threshold);
 
+	/**
+	 * Sets whether the massindexer should try to merge types from the same entity hierarchy into groups when performing
+	 * loading operations or have a single group for each type.
+	 * <p>
+	 * Defaults to {@code true}.
+	 *
+	 * @param merge Whether to merge types into groups.
+	 * @return {@code this} for method chaining
+	 */
+	@Incubating
+	MassIndexer mergeTypeGroups(boolean merge);
```

Comment:
So... technically, this is already controlled by whether those types have compatible loading strategies (according to `equals`), *and* by `org.hibernate.search.mapper.pojo.loading.spi.PojoMassLoadingStrategy#groupingAllowed`.

I'd argue that if there's a reason not to merge two types together... they should probably just have distinct loading strategies, or at least strategies that return the correct value from `groupingAllowed`?

Also, the same problem exists for selection loading (search), where merging is also based on whether loading strategies are equal.

TBH I'm not entirely sure this is even necessary, especially after your fix. And if it is... I'd rather get a bug report to explain to me why it is, so that we can think of an automated solution.

For example we might need something that _automatically_ sets the loading strategy not just by entity hierarchy, but by something finer grained... the ORM "table mapping" for example?
E.g. entities with `InheritanceType.TABLE_PER_CLASS` automatically get distinct strategies, those with `InheritanceType.SINGLE_TABLE` automatically get a shared strategy, and those with `InheritanceType.JOINED` will get a shared strategy, but one which would only enable grouping when targeting all concrete subtypes of a common entity supertype.
That, of course, would definitely be a separate Jira though.